concent_api/api_testing_common.py

```diff
@@ -185,21 +206,24 @@ def _print_data(data: Optional[Union[bytes, str]], url: str) -> None:
             print('SEND ({})'.format(url))
             print('MESSAGE:')
             print_golem_message(data)
-
+    if headers is None:
+        headers = REQUEST_HEADERS
     assert all(value not in ['', None] for value in [endpoint, host, REQUEST_HEADERS])
     url = "{}/api/v1/{}/".format(host, endpoint)
-
     _print_data(data, url)
-    response = requests.post("{}".format(url), headers=REQUEST_HEADERS, data=_prepare_data(data), verify=False)
+    response = requests.post("{}".format(url), headers=headers, data=_prepare_data(data), verify=False)
     _print_response(private_key, public_key, response)
     validate_response_status(response.status_code, expected_status)
     validate_content_type(response.headers['Content-Type'], expected_content_type)
     validate_response_message(response.content, expected_message_type, private_key, public_key)
+    if 'Concent-Golem-Messages-Version' in response.headers:
+        validate_golem_version(response.headers['Concent-Golem-Messages-Version'], expected_golem_version)
     print()
     content_type = response.headers['Content-Type']
     if 'text/html' in content_type:
         return None
-    elif content_type == 'json':
+    elif content_type == 'application/json':
+        validate_error_code((json.loads(response.content))['error_code'], expected_error_code)
```

Comment:
Same here.

Reply:
Well yes, and it's the only time we can get `json` response from concent, so I thought it's all good?

Reply:
Please add `and expected_error_code is not None` to condition, that is what I have meant by 'same here'. Then please remove `if` from `validate_error_code` function.

About the error codes, I couldn't find a place when we set something else than our custom error codes in response, that is why I was asking about it. If you are sure there is something like `bad-response` in `error_code` field it would be good to know why ;) I currently don't know.

Reply:
Now that You mention that, `bad-request` and `not-found` might be some default strings.
``` python
Running TC: test_case_ping_message_send_to_concent_with_malformed_golem_messages_should_get_http400_response_with_bad_request_error
SEND (http://devel.concent.golem.network/api/v1/send/)
MESSAGE:
    type                           = Ping
    timestamp                      = 1542196028
    encrypted                      = False
    sig                            = None
    header                         = MessageHeader(type_=1001, timestamp=1542196028, encrypted=False)
STATUS: 400 Bad Request
{'error_message': 'Bad Request', 'error_code': 'bad-request'}

Running TC: test_case_ping_message_send_to_concent_with_unsupported_golem_messages_should_get_http404_response_with_not_found_error
SEND (http://devel.concent.golem.network/api/v1/send/)
MESSAGE:
    type                           = Ping
    timestamp                      = 1542196028
    encrypted                      = False
    sig                            = None
    header                         = MessageHeader(type_=1001, timestamp=1542196028, encrypted=False)
STATUS: 404 Not Found
{'error_message': 'Not Found', 'error_code': 'not-found'}
```

Reply:
Done